lib/playlistGenerator/PlaylistGenerator.js

```diff
@@ -233,9 +233,9 @@ var doPersist =  function() {
     that.emit('beforePlaylistChanged', that);
 
     that.isDirty = false;
-    if(that.playlistImp.inner.liveWindowDuration) {
-        that.playlistImp.inner.presentationEndTime = that.playlistImp.getCurrentTime() + that.playlistImp.inner.liveWindowDuration;
-    }
+    that.playlistImp.inner.presentationEndTime = that.playlistImp.getCurrentTime() + that.playlistImp.inner.liveWindowDuration;
```

Comment:
When the entry is stopped the PlaylistGenerator object is removed. The entry is added to crossDCStreamPersistence and the business logic of stopped and ended is only handled there. The reason there is code for presentationEndTime and expirationTime values in PlaylistGenerator is because you requested that these properties  will always have value. It's by purpose that as long as the entry is live both have future value. As soon as the entry is stopped the crossDCStreamPersistence changes the expiration time to now and this causes the packager to return 404. when entry is not provided by any DC the presentationEndTime is set to now and packager puts END_LIST. I checked that it works.